hawkular-alerts-engine-extensions/hawkular-alerts-aggregated-events-alerter/src/main/java/org/hawkular/alerts/extensions/ExpressionManager.java

```diff
@@ -0,0 +1,181 @@
+
+                Collection<FullTrigger> activeTriggers = new ArrayList<>();
+                for (Trigger trigger : triggers) {
+                    Collection<Condition> conditions = null;
+                    List<Condition> activeConditions = new ArrayList<>();
+                    try {
+                        if (trigger.isEnabled()) {
+                            conditions = definitions.getTriggerConditions(trigger.getTenantId(), trigger.getId(), null);
+                            log.infof("Checking [%s] Conditions for enabled trigger [%s]!", conditions.size(),
+                                    trigger.getName());
+                        }
+                    } catch (Exception e) {
+                        log.error("Failed to fetch Conditions when scheduling metrics conditions for " + trigger, e);
+                        continue;
+                    }
+                    if (null == conditions) {
+                        continue;
+                    }
+                    for (Condition condition : conditions) {
+                        if (condition instanceof ExternalCondition) {
+                            ExternalCondition externalCondition = (ExternalCondition) condition;
+                            if (TAG_VALUE.equals(externalCondition.getAlerterId())) {
+                                activeConditions.add(externalCondition);
+                            }
+                        }
+                    }
+                    if (!activeConditions.isEmpty()) {
+                        FullTrigger activeTrigger = new FullTrigger();
+                        activeTrigger.setTrigger(trigger);
+                        activeTrigger.setConditions(activeConditions);
+                        activeTriggers.add(activeTrigger);
+                    }
+                }
+                if (!activeTriggers.isEmpty()) {
+                    log.infof("ActiveTriggers: %s", activeTriggers);
+                    cep.updateConditions(activeTriggers);
+                }
+            } catch (Exception e) {
+                log.error("Failed to fetch Triggers for external conditions.", e);
+            }
+        });
+    }
+
+    @Lock(LockType.READ)
+    @Override
+    public TreeSet<Event> processEvents(TreeSet<Event> events) {
+        if (isEmpty(events)) {
+            return events;
+        }
+        List<Event> retained = new ArrayList<>();
+        TreeSet<Event> filtered = new TreeSet<>();
+        for (Event event : events) {
+            if (event.getTags().get(TAG_NAME) != null && event.getTags().get(TAG_NAME).equals(TAG_VALUE)
+                    && event.getContext().get(CONTEXT_PROCESSED) == null) {
+                retained.add(event);
+            } else {
+                filtered.add(event);
+            }
+        }
+        if (!retained.isEmpty()) {
+            executor.submit(() -> cep.processEvents(retained));
+        }
+        return filtered;
+    }
```

Comment:
The fact is that given the jdoc and the impl I still found it quite confusing so I do think something needs to done for clarity.   In general extensions are going to generate some set of forwarded data/events, and it may make it more clear (certainly for me, maybe for others) to make that more explicit in the jdoc and/or impl.

Also, CONTEXT_PROCESSED could use some short inline comment, I think.  Will an event with that context ever be forwarded/persisted?  I don't think so, but if so, it would be nice if that context was scrubbed.